test/integration/composite-weighted/composite-weighted.test.ts

```diff
@@ -0,0 +1,30 @@
+import { calculateMeasureReports } from '../../../src/calculation/Calculator';
+import { getJSONFixture } from '../helpers/testHelpers';
+import { CompositeMeasureReport } from '../../../src/calculation/CompositeReportBuilder';
+
+const MEASURE_BUNDLE: fhir4.Bundle = getJSONFixture('composite-weighted/composite-weighted-bundle.json');
+const PATIENT_COMP1_NUMER_COMP2_NUMER: fhir4.Bundle = getJSONFixture(
+  'composite-weighted/patients/patient-comp1-numer-comp2-numer-bundle.json'
+);
+const PATIENT_COMP1_DENOM_COMP2_NUMER: fhir4.Bundle = getJSONFixture(
+  'composite-weighted/patients/patient-comp1-denom-comp2-numer-bundle.json'
+);
+
+describe('Composite measure weighted scoring', () => {
+  it('should calculate weighted result for components', async () => {
+    const { results } = await calculateMeasureReports(
+      MEASURE_BUNDLE,
+      [PATIENT_COMP1_NUMER_COMP2_NUMER, PATIENT_COMP1_DENOM_COMP2_NUMER],
+      {
+        measurementPeriodStart: '2023-01-01',
+        measurementPeriodEnd: '2023-12-31',
+        reportType: 'summary'
+      }
+    );
+
+    expect(results).toBeDefined();
+    const result = results as CompositeMeasureReport;
+    expect(result.group[0].measureScore?.value).toBeDefined();
+    expect(result.group[0].measureScore?.value).toEqual(0.625);
```

Comment:
Thoughts on adding a check on the numerator/denominator counts as well? Would be something like the following:

```javascript
expect(result.group[0].population?.find(e => e.code?.coding?.[0].code === 'denominator')?.count).toEqual(4);
expect(result.group[0].population?.find(e => e.code?.coding?.[0].code === 'numerator')?.count).toEqual(2.5);
```

For weighted scoring, I feel like it's rare that the numer/denom counts are different from what's expected but still yield the correct measure score. But I have seen this happen a lot when testing all-or-nothing and opportunity scoring (ex. opportunity should give numer of 3, denom of 9, and measure score of 1/3, but instead it incorrectly outputs numer of 1, denom of 3, and measure score of 1/3. You wouldn't know that the numer/denom counts are incorrect just by checking the measure score). 

Might be good to have these additional checks in place so that when we expand upon our integration testing for other scoring types we can have a consistent approach across all scoring types (i.e. check that measure score is defined, measure score is correct, and numer/denom counts are both correct).

Reply:
Oh sure- I like this idea. Thanks for the suggestion!

Reply:
Since it's a composite report, do you even need to do a `.find`? We enforce that the populations is an ordered tuple of [denomPop, numerPop], so you _could_ just access those instead. The test will break if we ever change that, but that's expected.